upup/pkg/fi/nodeup/command.go

```diff
@@ -477,14 +477,19 @@ func evaluateSpec(c *NodeUpCommand, nodeupConfig *nodeup.Config) error {
 	return nil
 }
 
-func evaluateHostnameOverride(cloudProvider api.CloudProviderID) (string, error) {
+func evaluateHostnameOverride(cloudProvider api.CloudProviderID, useInstanceIDForNodeName bool) (string, error) {
 	switch cloudProvider {
 	case api.CloudProviderAWS:
-		hostnameBytes, err := vfs.Context.ReadFile("metadata://aws/meta-data/local-hostname")
+		source := "local-hostname"
+		if useInstanceIDForNodeName {
+			source = "instance-id"
+		}
+		nodeNameBytes, err := vfs.Context.ReadFile("metadata://aws/meta-data/" + source)
 		if err != nil {
-			return "", fmt.Errorf("error reading local-hostname from AWS metadata: %v", err)
+			return "", fmt.Errorf("error reading %s from AWS metadata: %v", source, err)
 		}
-		return string(hostnameBytes), nil
```

Comment:
If we use the instance ID we should not return it here. That would cause hostname override to be set on k8s components, which we want to avoid when we can.

Reply:
As far as my IDE can see, this goes into kubelet and kube-proxy. Both use it as the node name. So I don't see how we can avoid either.

Reply:
`NodeName` has a fallback to `os.hostname` which is also what kubelet and kube-proxy use as default.

Reply:
Right. I suggest an additional test for RBN and if so return empty string.

Reply:
I'm failing to see the benefit of adding such logic.

Reply:
The whole motivation for using just the hostname was to not set hostname override. It's one of the failing CIS checks we have from https://github.com/kubernetes/kops/issues/6150.

Reply:
Ah, another silly, inappropriate CIS benchmark thing.

But can we really remove the override? The hostname would be a FQDN, whereas the CCM only takes an unqualifed instance ID.

Reply:
Yes. The CIS benchmark is quite silly. But you know how it is with compliance, I am sure.

The fallback to `os.hostname()` would produce only the instance ID, not FQDN.

Reply:
I think a good follow-up PR would be to return nothing if `os.hostname()` matches what this function would otherwise return.

Reply:
I can do that.